src/modules/Elsa.MassTransit.AzureServiceBus/Features/AzureServiceBusFeature.cs

```diff
@@ -31,18 +53,81 @@ public override void Configure()
         {
             massTransitFeature.BusConfigurator = configure =>
             {
+                var consumers = massTransitFeature.GetConsumers().ToList();
+                var shortLivedConsumers = consumers
+                    .Where(c => c.IsShortLived)
+                    .ToList();
+                RegisterConsumers(consumers);
+                
                 configure.AddServiceBusMessageScheduler();
+                configure.AddConsumers(shortLivedConsumers.Select(c => c.ConsumerType).ToArray());
                 
                 configure.UsingAzureServiceBus((context, serviceBus) =>
                 {
+                    var options = context.GetRequiredService<IOptions<MassTransitWorkflowDispatcherOptions>>().Value;
+                    var instanceNameRetriever = context.GetRequiredService<IInstanceNameRetriever>();
+
                     if (ConnectionString != null) 
                         serviceBus.Host(ConnectionString);
-                    
                     serviceBus.UseServiceBusMessageScheduler();
                     ConfigureServiceBus?.Invoke(serviceBus);
+
+                    foreach (var consumer in shortLivedConsumers)
+                    {
+                        serviceBus.ReceiveEndpoint($"Elsa-{instanceNameRetriever.GetName()}-{consumer.Name}", configurator =>
```

Comment:
If this name becomes too long for subscriptions, we could consider generating a hash of fixed length. Or, a part of the endpoint could be hashed.